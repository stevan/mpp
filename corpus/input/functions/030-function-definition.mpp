sub add($x, $y) {
    return $x + $y;