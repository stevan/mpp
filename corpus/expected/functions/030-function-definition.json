[
  {
    "type": "Sub",
    "name": "add",
    "parameters": [
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$x"
        }
      },
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$y"
        }
      }
    ],
    "body": [
      {
        "type": "Return",
        "value": {
          "type": "BinaryOp",
          "operator": "+",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Variable",
            "name": "$y"
          }
        }
      }
    ]
  }
]